### ISSUE TITLE
Duplicate / confusing jars in shipped game zip

### ISSUE THREAD

**Issue by Cervator** [MovingBlocks/Terasology#4671]
On semi-randomly poking around I noticed by change that our game zip ships with a few dependencies that look to be duplicated with different versions. This could have snuck in and not matter, could be transitive dependencies etc, but it seemed worth mentioning since it includes big scary ole Reflections.

Suspicious sets:

* `annotations-13.0.jar` + `annotations-2.0.1.jar`
*

**Comment by keturn** [comment]
> reflections-0.9.10.jar + reflections-0.9.12-MB.jar - I think the second one might be built via our fork?

Correct. And this is the one that worries me. On the one hand, if by some chance we have some dependency with a transitive dependency on the upstream reflections-0.9.10, it might be a little bloat but I don't know if it's risky beyond that.

But the Reflections class is a part of the public interface of gestalt-module. So if we somehow end up with multiple different implementations of Reflections that we're trying to pass around among engine, gestalt-module, and nui.reflect or whatever else, _that_ sounds like it might make for some very hard-to-debug situations.

I guess the thing to do is look at the dependency graph in a build scan and see what is depending on each of them.

**Comment by soloturn** [comment]
Picked this up investigating an unrelated Android-build-verification issue on TeraNUI ([MovingBlocks/TeraNUI#83](https://github.com/MovingBlocks/TeraNUI/pull/83)) — can answer the `reflections-0.9.10.jar` + `reflections-0.9.12-MB.jar` question directly.

**Yes, `reflections-0.9.12-MB.jar` is our fork** ([MovingBlocks/reflections](https://github.com/MovingBlocks/reflections), forked from `ronmamo/reflections` at its `0.9.12-SNAPSHOT`). It carries exactly one commit on top of upstream, from 2019: it drops the `javax.servlet:servlet-api` and `org.slf4j:slf4j-simple` optional dependencies (and the two `ClasspathHelper` methods — `forWebInfLib`/`forWebInfClasses` — that needed `ServletContext`), fixes a couple of Windows-only test failures, and renames the Maven coordinates from `org.reflections:reflections` to `org.terasology:reflections`. No change to actual scanning/reflection behavior. I checked whether upstream has since matched this independently — it hasn't; current upstream (`0.10.2`) still carries `servlet-api` and the WEB-INF helpers unchanged.

**Why both jars ship**: it's a real inconsistency, not a transitive-dependency accident. `gestalt-module` depends on the fork, but TeraNUI's `nui`, `nui-reflect`, `nui-gestalt5`, and `nui-gestalt7` modules each declared a *direct* dependency on plain upstream `org.reflections:reflections:0.9.10` — so both land on the same classpath and both get shipped. This is invisible to a normal Gradle/JVM build (classpath just has two jars, nothing complains), which is presumably why it's sat for years — it only becomes a hard failure under Android's `checkDuplicateClasses` task, which is what actually surfaced it for me while building a real Android verification APK.

**Fixed on the TeraNUI side**: [MovingBlocks/TeraNUI#83](https://github.com/MovingBlocks/TeraNUI/pull/83) migrates `nui`/`nui-reflect`/`nui-gestalt5`/`nui-gestalt7` to depend on the `org.terasology:reflections:0.9.12-MB` fork directly instead of upstream — same behavior, so it's a zero-risk fix. Once that merges and rolls into an engine build, `reflections-0.9.10.jar` should stop appearing in the shipped zip alongside the fork.

**On "should we drop the fork entirely and go back to upstream"**: I looked into that too. Current upstream `0.10.2` removes `Store.getAll()` (used by `TypeRegistry.getSubtypesOf`) and changes `SubTypesScanner`'s default `java.lang.Object`-exclusion behavior — `TypeRegistry` explicitly relies on the non-default `new SubTypesScanner(false)`, which doesn't map cleanly onto the new `Scanners.SubTypes` enum (its exclusion filter is baked in, not exposed as a constructor flag anymore). Migrating off the fork to modern upstream is a real, careful piece of work with actual behavioral risk to type discovery across the engine/module ecosystem — not something to fold into this cleanup casually. Left it on the fork for now; happy to open a separate issue for the upstream-0.10.2 migration if there's appetite for it.
